src/Bindings/PluginLua.cpp

```diff
@@ -72,12 +72,13 @@ void cPluginLua::Close(void)
 	cResettablePtrs resettables;
 	std::swap(m_Resettables, resettables);
 	{
-		cCSUnlock Unlock(Lock);
+		m_CriticalSection.Unlock();
 		for (auto resettable: resettables)
 		{
 			resettable->Reset();
 		}
 		m_Resettables.clear();
+		m_CriticalSection.Lock();
```

Comment:
This will fail hard if anything between `Unlock()` and `Lock()` exits via exception. The RAII for unlocking is needed.